pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,178 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+const (
+	defaultGitTimeout = 20 * time.Second
```

Comment:
Nit: given this constant also exists in `pkg/ctl/generate/profile.go`, should we move it elsewhere, e.g. within the `git` package and only use one?